aws/resource_aws_servicecatalog_product.go

```diff
@@ -0,0 +1,293 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+
+	"github.com/aws/aws-sdk-go/service/servicecatalog"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsServiceCatalogProduct() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsServiceCatalogProductCreate,
+		Read:   resourceAwsServiceCatalogProductRead,
+		Update: resourceAwsServiceCatalogProductUpdate,
+		Delete: resourceAwsServiceCatalogProductDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"created_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"distributor": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"owner": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"product_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"product_type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"support_description": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"support_email": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"support_url": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"artifact": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"description": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"load_template_from_url": {
+							Type:     schema.TypeString,
+							Required: true,
+							DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+								// Only check for a diff on initial create.
+								if d.Id() != "" {
+									return true
+								}
+								return false
+							},
```

Comment:
We can in fact source the URL from the API, so we shouldn't need the suppression function here:
https://docs.aws.amazon.com/servicecatalog/latest/dg/API_DescribeProvisioningArtifact.html#servicecatalog-DescribeProvisioningArtifact-response-Info